submissions/JuliaNovak/tiny-js-world/index.js

```diff
@@ -0,0 +1,76 @@
+// ======== OBJECTS DEFINITIONS ========
+// Define your objects here
+class Inhabitant {
+   constructor(species, name, gender, legs, hands, saying) {
+      this.species = species;
+      this.name = name;
+      this.gender = gender;
+      this.legs = legs;
+      this.hands = hands;
+      this.saying = saying;
+      this.properties = [
+         "name",
+         "species",
+         "gender",
+         "legs",
+         "hands",
+         "saying",
+      ];
+   }
+
+   get props() {
+      return this.properties.map((value) => this[value]);
+   }
+
+   display(inhabitantsArr) {
+      inhabitantsArr.map((inhabitant) => {
+         print(inhabitant.props);
+      });
+   }
+}
+
+class Human extends Inhabitant {
+   constructor(species, name, gender, saying) {
+      super(species, name, gender);
+      this.legs = 2;
+      this.hands = 2;
+      this.saying = saying;
+   }
+}
+
+class Cat extends Inhabitant {
+   constructor(species, name, gender, saying) {
+      super(species, name, gender);
+      this.legs = 4;
+      this.hands = 0;
```

Comment:
OOP helps us to skip properties that objects do not need.
Imagine we have a complete bio taxonomy. Making every living being have properties that they do not have by natural design clutters them and forces us to handle special cases.